builtin/rebase--helper.c

```diff
@@ -83,8 +111,28 @@ int cmd_rebase__helper(int argc, const char **argv, const char *prefix)
 		return !!sequencer_continue(&opts);
 	if (command == ABORT && argc == 1)
 		return !!sequencer_remove_state(&opts);
-	if (command == MAKE_SCRIPT && argc > 1)
-		return !!sequencer_make_script(stdout, argc, argv, flags);
+	if (command == MAKE_SCRIPT && (argc == 1 || argc == 2)) {
+		struct argv_array f_argv = ARGV_ARRAY_INIT;
+
+		if (!upstream && squash_onto)
+			write_file(path_squash_onto(), "%s\n", squash_onto);
+
+		ret = get_revision_ranges(upstream, onto, &head_hash, &revisions);
+		if (ret)
+			return ret;
+
+		argv_array_push(&f_argv, argv[0]);
+		argv_array_push(&f_argv, revisions);
+		if (argc == 2)
+			argv_array_push(&f_argv, argv[1]);
```

Comment:
`restrict_revision` is the only thing that is not parsed via parse options. And that is the reason why the code here talks about `argv[1]` which is not as obvious as `restrict_revision` would be...

Reply:
Same as above, I think a comment would be more appropriate than a variable.

Reply:
Same as above, you are wrong here. A code comment is *always* more prone to become stale than real code, as the code comment is ignored by the compiler, and the code simply is not.

Plus, I have no idea why you insist on passing the `restrict_revision` as a non-option. It would make everything so much clearer, and solve the current problem, if you simply did it via `--restrict-revision=...` instead of playing games with positional parameters.